quipucords/fingerprinter/task.py

```diff
@@ -360,8 +373,9 @@ def _process_details_report(self, manager_interrupt, details_report):
                                      fingerprint_dict.get('products', []),
                                      fingerprint_dict.get('is_redhat'))}
                         insights_host['facts'] = [facts]
-                        insights_host['system_profile'] = \
-                            self.format_system_profile(nested_facts)
+                        if bool(nested_facts):
```

Comment:
using bool is fine here but unnecessary, you also can just check `if nested_facts`. If you like bool for readability then leave it 😄

Reply:
I thought about that. I wasn't sure whether `if nested_facts` returns `False` when the dictionary is empty or when the variable does not exist.

Reply:
I just checked. You are right. :-D